src/librustc_codegen_llvm/back/write.rs

```diff
@@ -777,12 +776,10 @@ pub(crate) unsafe fn codegen(
             }
         }
 
-        if config.obj_is_bitcode {
-            if config.emit_obj {
-                debug!("copying bitcode {:?} to obj {:?}", bc_out, obj_out);
-                if let Err(e) = link_or_copy(&bc_out, &obj_out) {
-                    diag_handler.err(&format!("failed to copy bitcode to object file: {}", e));
-                }
+        if config.emit_obj == EmitObj::Bitcode {
+            debug!("copying bitcode {:?} to obj {:?}", bc_out, obj_out);
+            if let Err(e) = link_or_copy(&bc_out, &obj_out) {
+                diag_handler.err(&format!("failed to copy bitcode to object file: {}", e));
             }
 
             if !config.emit_bc {
```

Comment:
FWIW I think this is something that could eventually use some cleanup to. It'd be kinda nice if we had something like a dynamic set of "here's a bunch of temporary files that we used" and they're all automatically always removed unless explicitly whitelisted by something else in the compiler to keep it. 

That way the fact that `EmitObj::Bitcode` needs bitcode means that it would be produced early on (and then asserted that we produced it here), but we wouldn't have to have all these sprawling "oh delete the file if we didn't actually ask for it as an output" statements.

Reply:
I agree it would be nice, but it feels like scope creep for this PR so I won't do it for now.